framework/include/executioners/Steady.h

```diff
@@ -56,12 +56,31 @@ class Steady : public Executioner
    * Set a general iteration number for the purpose of outputting, useful in the presence of a
    * nested solve This output iteration number may be set by the parent app for a sub-app, e.g.
    * OptimizeSolve.
+   *
+   * @param iteration_number The iteration number.
    */
   virtual void setIterationNumberOutput(unsigned int iteration_number)
   {
     _output_iteration_number = iteration_number;
   }
 
+  /**
+   * Set whether we are solving an optimization problem via OptimizeSolve.
+   *
+   * @param is_optimize_solve Whether the system is performing optimization.
+   */
+  virtual void setOptimizationFlag(const bool is_optimize_solve)
```

Comment:
We need to be able to not have that

I know you are trying to follow previous guidance but you found a way around this (no optim-specific code in framework, instead stuff usable for other purposes as well) for exodus and we need to be using the same mechanic